elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,213 @@
+          break;
+       case KEYCODE.UP:
+         // If arrow up, move to previous item. Wrap if necessary.
+         event.preventDefault();
+         let prev = this._previousMenuItem(el) || this._lastMenuItem;
+         prev.focus();
+         break;
+       case KEYCODE.ESCAPE:
+         // If escape, close the menu.
+         event.preventDefault();
+         this.opened = false;
+         break;
+       case KEYCODE.TAB:
+         // If escape, close the menu.
+         event.preventDefault();
+         this.opened = false;
+         break;
+       default:
+         break;
+      }
+      // If letter key, move to an item which starts with that letter.
+      if (event.keyCode > 64 && event.keyCode < 91) {
+        for (let i = 0, child; child = this.children[i]; i++) {
+          if (child.innerText.trim()[0] === event.key) {
+            event.preventDefault();
+            child.focus();
+            break;
+          }
+        }
+      }
+    }
+
+    open() {
+      this._setTabindex(true);
+      this._firstMenuItem.focus();
+    }
+
+    set opened(isOpened) {
+      if (!!isOpened) {
+        this.setAttribute('opened', '');
+        this.open();
+      } else {
+        this.removeAttribute('opened');
+        this._setTabindex(false);
+      }
+    }
+
+    get opened() {
+      return this.hasAttribute('opened');
+    }
+
+    _upgradeProperty(prop) {
+      if (this.hasOwnProperty(prop) || this.hasAttribute(prop)) {
+        let value = this[prop];
+        delete this[prop];
+        this[prop] = value;
+      }
+    }
+
+    _onFocusOut(event) {
+      if (event.relatedTarget === null || !this.contains(event.relatedTarget)) {
+        this.opened = false;
+      }
+    }
+
+    /**
+     * `connectedCallback` fires when the element is inserted into the DOM.
+     * It's a good place to set the initial `role`, `tabindex`, internal state,
+     * and install event listeners.
+     */
+    connectedCallback() {
+      if (!this.hasAttribute('role')) {
+        this.setAttribute('role', 'menu');
+      }
+
+      // A user may set a property on an _instance_ of an element,
+      // before its prototype has been connected to this class.
+      // The `_upgradeProperty` method will check for any instance properties
+      // and run them through the proper class setters.
+      // See the [lazy properites](#lazy-properties) section for more details.
+      this._upgradeProperty('opened');
+
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('focusout', this._onFocusOut);
+    }
+
+    /**
+     * Unregisters the event listeners that were set up in `connectedCallback`.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('keydown', this._onKeyDown);
+      this.removeEventListener('focusout', this._onKeyDown);
+    }
+
+    /**
+     * Resets the menu by removing tabindex on all menu items.
+     */
+    _setTabindex(set) {
+      const children = Array.from(this.children);
```

Comment:
You can call `forEach` on `NodeList`, so `Array.from()` is not necessary. You could also do a `for...of` loop.